diff --git a/scripts/wsrep_sst_mariabackup.sh b/scripts/wsrep_sst_mariabackup.sh
index 4eb9917d341..d2d3c0003ce 100644
--- a/scripts/wsrep_sst_mariabackup.sh
+++ b/scripts/wsrep_sst_mariabackup.sh
@@ -1037,6 +1037,19 @@ then
     if [ ! -r "${STATDIR}/${IST_FILE}" ]
     then
 
+        if [ ${DISABLE_SST:=0} -eq 1 ]
+        then
+            wsrep_log_error "#####################################################################################"
+            wsrep_log_error "SST disabled due to danger of data loss. Verify data and run the rejoin-unsafe errand"
+            wsrep_log_error "#####################################################################################"
+
+            if [ -n "${INTERRUPT_NOTIFY_CMD:=}" ]
+            then
+              ${INTERRUPT_NOTIFY_CMD}
+            fi
+            exit 1
+        fi
+
         if [[ -d ${DATA}/.sst ]];then
             wsrep_log_info "WARNING: Stale temporary SST directory: ${DATA}/.sst from previous state transfer. Removing"
             rm -rf ${DATA}/.sst
